libs/parallelism/executors/include/hpx/executors/scheduler_executor.hpp

```diff
@@ -0,0 +1,319 @@
+            };
+        }
+
+        template <typename F, typename... Ts>
+        auto captured_args_then(F&& f, Ts&&... ts)
+        {
+            return [f = std::forward<F>(f),
+                       t = hpx::make_tuple(std::forward<Ts>(ts)...)](
+                       auto i, auto&& predecessor, auto& v) mutable {
+                v[i] = hpx::util::invoke_fused(
+                    hpx::util::bind_front(std::forward<F>(f), i,
+                        std::forward<decltype(predecessor)>(predecessor)),
+                    std::move(t));
+            };
+        }
+
+        template <typename F, typename... Ts>
+        auto captured_args_then_void(F&& f, Ts&&... ts)
+        {
+            return [f = std::forward<F>(f),
+                       t = hpx::make_tuple(std::forward<Ts>(ts)...)](
+                       auto i, auto&& predecessor) mutable {
+                hpx::util::invoke_fused(
+                    hpx::util::bind_front(std::forward<F>(f), i,
+                        std::forward<decltype(predecessor)>(predecessor)),
+                    std::move(t));
+            };
+        }
+#endif
+    }    // namespace detail
+
+    ///////////////////////////////////////////////////////////////////////////
+    // A scheduler_executor wraps any P2300 scheduler and implements the
+    // executor functionalities for those.
+    template <typename BaseScheduler>
+    struct scheduler_executor
+    {
+        template <typename Scheduler,
+            typename Enable = std::enable_if_t<
+                hpx::execution::experimental::is_scheduler_v<Scheduler>>>
+        constexpr explicit scheduler_executor(Scheduler&& sched)
+          : sched_(std::forward<Scheduler>(sched))
+        {
+        }
+
+        /// \cond NOINTERNAL
+        constexpr bool operator==(scheduler_executor const& rhs) const noexcept
+        {
+            return sched_ == rhs.sched_;
+        }
+
+        constexpr bool operator!=(scheduler_executor const& rhs) const noexcept
+        {
+            return sched_ != rhs.sched_;
+        }
+
+        constexpr auto const& context() const noexcept
+        {
+            return *this;
+        }
+        /// \endcond
+
+        /// \cond NOINTERNAL
+        // Associate the parallel_execution_tag executor tag type as a default
+        // with this executor.
+        using execution_category = parallel_execution_tag;
+
+        // Associate the static_chunk_size executor parameters type as a default
+        // with this executor.
+        using executor_parameters_type = static_chunk_size;
+
+        template <typename T, typename... Ts>
+        using future_type = hpx::future<T>;
+
+        // NonBlockingOneWayExecutor interface
+        template <typename F, typename... Ts>
+        void post(F&& f, Ts&&... ts)
+        {
+            detach(transform(schedule(sched_),
+                detail::captured_args(
+                    std::forward<F>(f), std::forward<Ts>(ts)...)));
```

Comment:
Is `capture_args` different than `deferred_call`? This question applies to all the captures below as well.

Reply:
The `captured_args` helper is very similar to `deferred_call`, indeed. It however inserts an extra argument into the embedded function call. Those could be replaced with something like `bind(f, _1, ts...)` - I'll try that.

Reply:
FWIW, in order to use bind() we need to merge #5489 first.